src/net/polkadot.ts

```diff
@@ -82,32 +84,24 @@ export default class Polkadot extends Api {
     polkadotApi: ApiPromise,
     polkadotAddress: string,
     asset: Asset,
-  ): Promise<string> {
+  ): Promise<AssetBalance> {
     try {
-      if (polkadotApi) {
-        // check if the asset is the native DOT asset and return DOT balance
-        if (isDot(asset)) {
-          const balance = await Polkadot.getGasCurrencyBalance(polkadotApi, polkadotAddress);
-          return balance;
-        }
-
-        let ethAssetID = polkadotApi.createType('AssetId', 'ETH');
-        // create asset ID for tokens
-        if (isErc20(asset)) {
-          ethAssetID = polkadotApi.createType('AssetId', { Token: asset.address });
-        }
-        if (polkadotAddress) {
-          const balance = await polkadotApi.query.assets.balances(
-            ethAssetID,
-            polkadotAddress,
-          );
-
-          return balance.toString();
+      let account: any;
+      // check if the asset is the native DOT asset and return DOT balance
+      if (isDot(asset)) {
+        const balance = await Polkadot.getGasCurrencyBalance(polkadotApi, polkadotAddress);
+        return { isAssetFound:false, amount:balance };
+      }
+      account = await polkadotApi.query.assets.account(PARACHAIN_ETHER_ASSET_ID, polkadotAddress);
```

Comment:
This code flow is still a bit confusing though, and we end up making two calls to `query.assets.account` if the asset is erc20, which is unnecessary.

So can we separate the body into into clear distinct cases using `if` statements:
- if DOT, fetch dot balances
- if Ether, fetch ether balances
- If Erc20, erc20 balances

Reply:
As suggested, I've modified the code.